src/components/Tags/MultiFilterStyleTagSelector.tsx

```diff
@@ -232,88 +412,32 @@ export function MultiFilterStyleTagSelector({
-                                      }
-                                      className="bg-white peer border-gray-200 data-[state=checked]:bg-primary-600 data-[state=checked]:text-primary-100 data-[state=checked]:border-primary-600 focus-visible:border-primary-600 focus-visible:ring-primary-500/50 size-4 shrink-0 rounded-[4px] border shadow-xs transition-shadow outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 h-4 w-4"
-                                    >
-                                      {isSelected(childTag) && (
-                                        <span className="flex items-center justify-center text-current transition-none">
-                                          <Check className="size-3.5" />
-                                        </span>
-                                      )}
-                                    </button>
-                                    <div className="flex items-center gap-2 flex-1">
-                                      <div className="h-3 w-3 rounded-full flex-shrink-0 border bg-green-100 border-green-300"></div>
-                                      <span className="text-sm">
-                                        {childTag.display}
-                                      </span>
-                                    </div>
-                                  </div>
-                                ))
-                              ) : (
-                                <div className="p-2 text-sm text-gray-500">
-                                  {t("no_tags")}
-                                </div>
-                              )}
-                            </PopoverContent>
-                          </Popover>
-                        </div>
-                      ))}
+                    {isMobile
+                      ? // Mobile tree view
+                        rootTags?.results
+                          ?.filter((tag) => tag.has_children)
+                          .map((tag) => (
+                            <TreeViewItem
+                              key={tag.id}
+                              tag={tag}
+                              selectedTags={selected}
+                              onTagToggle={handleSelect}
+                              resource={resource}
+                            />
+                          ))
+                      : // Desktop submenu view
+                        rootTags?.results
+                          ?.filter((tag) => tag.has_children)
+                          .map((tag) => (
+                            <GroupSubmenu
+                              key={tag.id}
+                              tag={tag}
+                              selectedTags={selected}
+                              onTagToggle={handleSelect}
+                              resource={resource}
+                              facilityId={facilityId}
+                            />
+                          ))}
```

Comment:
we could have achieved this without useBreakpoints and only using tailwind classes right? 🤔

Reply:
hmm yea, can circle back for a cleanup 👍